src/txn/txn.c

```diff
@@ -918,6 +917,7 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
          * Scan the history store for the given btree and key with maximum start timestamp to let
          * the search point to the last version of the key.
          */
+        F_SET(hs_cursor, WT_CURSTD_IGNORE_TOMBSTONE);
```

Comment:
Why do we need this?

Reply:
In case if there is a valid tombstone exists for the history store record, the search for the history store returns NOTFOUND because prepare transaction can see the tombstone that is visible. Due to this reason, we need to ignore the tombstone while retrieving the history store update. I can add a comment to explain the same.

Reply:
Understood. Yeah a comment would be good.